core/domain/exp_jobs_one_off_test.py

```diff
@@ -935,6 +935,261 @@ def test_no_action_is_performed_for_deleted_exploration(self):
+                    'content_1': {},
+                    'feedback_2': {},
+                    'hint_1': {},
+                    'content_2': {}
+                }
+            },
+            'interaction': {
+                'answer_groups': [],
+                'confirmed_unclassified_answers': [],
+                'customization_args': {},
+                'default_outcome': {
+                    'dest': 'Introduction',
+                    'feedback': {
+                        'content_id': 'feedback_2',
+                        'html': 'Correct Answer'
+                    },
+                    'param_changes': [],
+                    'refresher_exploration_id': None,
+                    'labelled_as_correct': True,
+                    'missing_prerequisite_skill_id': None
+                },
+                'hints': [{
+                    'hint_content': {
+                        'content_id': 'hint_1',
+                        'html': 'Hint 1'
+                    }
+                }],
+                'solution': {
+                    'correct_answer': {
+                        'ascii': 'x+y',
+                        'latex': 'x+y'
+                    },
+                    'answer_is_exclusive': False,
+                    'explanation': {
+                        'html': 'Solution explanation',
+                        'content_id': 'content_2'
+                    }
+                },
+                'id': 'MathExpressionInput'
+            },
+            'next_content_id_index': 0,
+            'param_changes': [],
+            'solicit_answer_details': False,
+            'classifier_model_id': None
+        }).to_dict()
+
+        self.save_new_exp_with_states_schema_v35(
+            self.NEW_EXP_ID, self.albert_id, {
+                'Introduction': states_dict
+            })
+
+        current_exp_schema_version = (
+            exp_domain.Exploration.CURRENT_EXP_SCHEMA_VERSION)
+        conversion_fn_name = (
+            '_convert_v%i_dict_to_v%i_dict' % (
+                current_exp_schema_version - 1,
+                current_exp_schema_version)
+        )
+
+        # Make a mock conversion function that raises an error.
+        mock_conversion = classmethod(
+            lambda cls, exploration_dict: exploration_dict['property_that_dne'])
+
+        with self.swap(
+            exp_domain.Exploration, conversion_fn_name, mock_conversion
+        ):
+            job_id = exp_jobs_one_off.ExplorationMigrationAuditJob.create_new()
+            exp_jobs_one_off.ExplorationMigrationAuditJob.enqueue(job_id)
+            self.process_and_flush_pending_tasks()
+
+            actual_output = (
+                exp_jobs_one_off.ExplorationMigrationAuditJob.get_output(
+                    job_id)
+            )
+
+            expected_output = [
+                u'[u\'MIGRATION_ERROR\', [u"Exploration exp_id1 failed migratio'
+                'n to v41: u\'property_that_dne\'"]]'
```

Comment:
Hm -- I think you should interpolate "41" with CURRENT_STATE_SCHEMA_VERSION, since otherwise anyone who does a migration is going to find out later on that this test mysteriously fails, which leads to wasted time.

Again, OK to do this in a future PR.

Reply:
Done